### ISSUE TITLE
no matching function for call to 'differentiate' on cuda -clad test

### ISSUE THREAD

**Issue by ioanaif** [vgvassilev/clad#205]
```
clad_cuda_test.cu:19:11: error: no matching function for call to 'differentiate'
    auto dfdx = clad::differentiate(f, "x");
          ^~~~~~~~~~~~~~~~~~~
./../clad/include/clad/Differentiator/Differentiator.h:148:4: note: candidate template ignored: could not match 'R (C::*)(Args...)' against 'double (*)(float *)'
   differentiate(R (C::*f)(Args...), ArgSpec args = "", const char* code = "") {
   ^
./../clad/include/clad/Differentiator/Differentiator.h:141:3: note: candidate function not viable: call to __host__ function from __global__ function
  differentiate(R (*f)(Args...), ArgSpec args = "", const char* code = "") {
  ^
1 error generated when compiling for sm_20.
```

********************************************

```
#include "../clad/include/clad/Differentiator/Differentiator.h"

#include <stdio.h>
#include <algorithm>
__device__ double f(float *x) {
  int i = blockIdx.x*blockDim.x + threadIdx.x;
  return x[i] * x[i] ;
}

__device__ double f_darg0(float *x);

__global__
void saxpy(int n, float *x)
{ float *y;
  auto dfdx = clad::differentiate(f, "x");
  int i = blockIdx.x*blockDim.x + threadIdx.x;
  if (i < n)
    y[i] =f_darg0(x);
}

int main(void)
{  
  int N = 1<<20;
  float *x, *y, *d_x, *d_y;
  x = (float*)malloc(N*sizeof(float));
  y = (float*)malloc(N*sizeof(float));

  cudaMalloc(&d_x, N*sizeof(float)); 
  cudaMalloc(&d_y, N*sizeof(float));

  for (int i = 0; i < N; i++) {
    x[i] = 1.0f;
    y[i] = 2.0f;
  }

  cudaMemcpy(d_x, x, N*sizeof(float), cudaMemcpyHostToDevice);
  cudaMemcpy(d_y, y, N*sizeof(float), cudaMemcpyHostToDevice);

  // Perform SAXPY on 1M elements
  saxpy<<<(N+255)/256, 256>>>(N, d_x);

  cudaMemcpy(y, d_y, N*sizeof(float), cudaMemcpyDeviceToHost);

  float maxError = 0.0f;
  for (int i = 0; i < N; i++)
    maxError = std::max(maxError, abs(y[i]-4.0f));
  printf("Max error: %f\n", maxError);

  cudaFree(d_x);
  cudaFree(d_y);
  free(x);
  free(y);
}
```

**Comment by vgvassilev** [comment]
@ioanaif, could you summarize what/how we fixed that? Of if it was a ill-formed example.

**Comment by ioanaif** [comment]
CUDA support which fixes this issue was added in PR #216  and  #232
